src/js/stores/AppsStore.js

```diff
@@ -146,12 +146,29 @@ function processApp(app) {
   return app;
 }
 
-function processApps(apps) {
-  return apps.map(function (app) {
-    return processApp(app);
+function processEmptyGroup(group) {
+  return Object.assign(processApp({id: group.id}), {
+    isGroup: true,
+    status: null
   });
 }
 
+function processAppsAndGroups(group, apps = []) {
```

Comment:
Thinking about it, I disagree regarding the "expected result" above. 

I think that the actual expected result should be one that matches the `v2/groups` endpoint in terms of structure (ie a tree of groups that may or may not contain apps). The "flattened" version as we have it right now (and as depicted in your expected result above) is only useful for searching and filtering. The data passed to the AppListComponent was "adapted" in this PR to avoid a deeper refactoring.
 
If we really want to improve this PR, then we should embrace the `groups` endpoint to the fullest, and adapt our code to reflect its tree-like structure. When we first wrote the AppListComponent, we modelled it according to the data returned by the `apps` endpoint – which was a flat list. Keeping this around makes no sense anymore.

cc @aldipower @orlandohohmeier @philipnrmn @Poltergeist 

So before I venture in this refactoring, please share your thoughts on the following:

1. Switch to the `GroupsStore` entirely. The `AppsStore.requestApps()` calls should be refactored to `GroupsStore.requestGroups()` and we should always refer to `groups`, rather than apps directly when requesting data. The meaning of this change is that `apps` themselves only exist in the context of a `group`. This behaviour is already reflected by our default AppListComponent rendering case now: it only renders applications found at the `currentGroup` level. 

2. For the other rendering case (search / filtering) we will add a flattened-list getter in the GroupsStore, so that we extract this responsibility away from the AppListComponent. Instead of populating this list inside the component (`AppListComponent.getGroupsFromApps()`), we should grab them via the `GroupsStore.apps` getter, which will return an array of apps and empty groups pretty much like it's currently done in this PR. Search and filtering operations may be performed against this list which should avoid expensive recursion and the likes. We finally expose this from the GroupsStore as it's our single source of truth.

3. We extract all the grouping-logic away from the AppListComponent: `.groupItems()`, `.getGroupsFromApps()` don't belong there.

4. Arguably, we should probably rename `AppListComponent` to `GroupedAppListComponent` (?)

5. Refactor all the tests... :(

Please share your ideas. Thanks!